source/extensions/filters/http/gzip/config.cc

```diff
@@ -12,27 +12,16 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Gzip {
 
-Server::Configuration::HttpFilterFactoryCb
-GzipFilterFactory::createFilter(const envoy::config::filter::http::gzip::v2::Gzip& proto_config) {
-  GzipFilterConfigSharedPtr config = std::make_shared<GzipFilterConfig>(proto_config);
-  return [config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
-    callbacks.addStreamFilter(std::make_shared<GzipFilter>(config));
-  };
-}
-
-Server::Configuration::HttpFilterFactoryCb
-GzipFilterFactory::createFilterFactory(const Json::Object&, const std::string&,
-                                       Server::Configuration::FactoryContext&) {
-  NOT_IMPLEMENTED;
-}
-
 Server::Configuration::HttpFilterFactoryCb
 GzipFilterFactory::createFilterFactoryFromProto(const Protobuf::Message& proto_config,
                                                 const std::string&,
                                                 Server::Configuration::FactoryContext&) {
-  return createFilter(
+  GzipFilterConfigSharedPtr config = std::make_shared<GzipFilterConfig>(
```

Comment:
Can `FactoryBase` provide a method that hides the downcast? What I would do is have the `GzipFilterFactory` implement `createFilterFactoryFromTypedProto`, and pass in the already downcast variant from `FactoryBase::createfilterFactoryFromProto`.